lib/iris/tests/unit/cube/test_CubeAttrsDict.py

```diff
@@ -0,0 +1,323 @@
+        attrs_1 = sample_attrs
+        attrs_2 = sample_attrs.copy()
+        getattr(attrs_2, aspect)["z"] = 99
+        assert attrs_1 != attrs_2
+        assert attrs_2 != attrs_1
+
+    def test_ne_locals_vs_globals(self):
+        attrs_1 = CubeAttrsDict(locals={"a": 1})
+        attrs_2 = CubeAttrsDict(globals={"a": 1})
+        assert attrs_1 != attrs_2
+        assert attrs_2 != attrs_1
+
+    def test_eq_dict(self):
+        # A CubeAttrsDict can be equal to a plain dictionary (which would create it)
+        vals_dict = {"a": 1, "b": 2, "history": "this"}
+        attrs = CubeAttrsDict(vals_dict)
+        assert attrs == vals_dict
+        assert vals_dict == attrs
+
+    def test_ne_dict_local_global(self):
+        # Dictionary equivalence fails if the local/global assignments are wrong.
+        # sample dictionary
+        vals_dict = {"title": "b"}
+        # these attrs are *not* the same, because 'title' is global by default
+        attrs = CubeAttrsDict(locals={"title": "b"})
+        assert attrs != vals_dict
+        assert vals_dict != attrs
+
+    def test_empty_not_none(self):
+        # An empty CubeAttrsDict is not None, and does not compare to 'None'
+        # N.B. this for compatibility with the LimitedAttributeDict
+        attrs = CubeAttrsDict()
+        assert attrs is not None
+        with pytest.raises(TypeError, match="iterable"):
+            # Cannot *compare* to None (or anything non-iterable)
+            # N.B. not actually testing against None, as it upsets black (!)
+            attrs == 0
+
+    def test_empty_eq_iterables(self):
+        # An empty CubeAttrsDict is "equal" to various empty containers
+        attrs = CubeAttrsDict()
+        assert attrs == {}
+        assert attrs == []
+        assert attrs == ()
+
+
+class TestDictOrderBehaviour:
+    def test_ordering(self):
+        attrs = CubeAttrsDict({"a": 1, "b": 2})
+        assert list(attrs.keys()) == ["a", "b"]
+        # Remove, then reinstate 'a' : it will go to the back
+        del attrs["a"]
+        attrs["a"] = 1
+        assert list(attrs.keys()) == ["b", "a"]
+
+    def test_globals_locals_ordering(self):
+        # create attrs with a global attribute set *before* a local one ..
+        attrs = CubeAttrsDict()
+        attrs.globals.update(dict(a=1, m=3))
+        attrs.locals.update(dict(f=7, z=4))
+        # .. and check key order of combined attrs
+        assert list(attrs.keys()) == ["a", "m", "f", "z"]
+        # create the "same" thing with locals before globals, *and* different key order
+        attrs = CubeAttrsDict()
+        attrs.locals.update(dict(z=4, f=7))
+        attrs.globals.update(dict(m=3, a=1))
+        # .. this shows that the result is not affected either by alphabetical key
+        # order, or the order of adding locals/globals
+        # I.E. result is globals-in-create-order, then locals-in-create-order
+        assert list(attrs.keys()) == ["m", "a", "z", "f"]
```

Comment:
Could you split this into two tests? I would rather we keep unit tests to each testing a single thing. Otherwise you run into the issue where you won't know whether the second test is failing if the first test fails.